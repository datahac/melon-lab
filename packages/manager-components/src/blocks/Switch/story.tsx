import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Switch from './index';

const data = {
  options: ['ETH-T-M', 'MLN-T-M'],
  labels: ['Buy', 'Sell'],
  name: 'name',
  value: 'value',
  onChange: action('changed'),
};

storiesOf('Blocks|Switch', module).add('Default', () => {
  return <Switch {...data} />;
});
