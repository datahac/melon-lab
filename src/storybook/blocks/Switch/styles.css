.switch {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.switch__wrapper {
  width: 100%;
  padding: 0 var(--space-m);
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.switch__icon {
  position: relative;
  height: 24px;
  width: 24px;
  background-color: var(--color-primary-dark);
  border-radius: 50%;
}

.switch__icon::before,
.switch__icon::after {
  font-family: Arial;
  font-size: var(--font-size-s);
  position: absolute;
  color: var(--color-white);
  left: 6px;
  line-height: 1;
  height: 9px;
}

.switch__icon::before {
  top: 2px;
  content: '←';
}

.switch__icon::after {
  bottom: 7px;
  content: '→';
}

.switch__wrapper {
  display: flex;
}

.switch__input {
  order: 2;
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.switch__icon {
  order: 3;
  flex: 0 1 24px;
  cursor: pointer;
  transition-duration: var(--transition-duration-default);
}

.switch__label {
  flex: 0 1 calc(100% / 2 - 24px);
}

.switch__label::before {
  position: relative;
}

.switch__input + .switch__label {
  order: 1;
  text-align: right;
  font-weight: var(--font-weight-bold);
}

.switch__input + .switch__label::before {
  content: attr(data-label-left) ' ';
}

.switch__input + .switch__label ~ .switch__label {
  order: 4;
  text-align: left;
  color: var(--color-grey);
}

.switch__input + .switch__label ~ .switch__label::before {
  content: attr(data-label-right) ' ';
}

.switch__input:checked + .switch__label::before {
  content: attr(data-label-right) ' ';
}

.switch__input:checked + .switch__label ~ .switch__label::before {
  content: attr(data-label-left) ' ';
}

.switch__input:checked ~ .switch__icon {
  transform: rotate(180deg);
}
