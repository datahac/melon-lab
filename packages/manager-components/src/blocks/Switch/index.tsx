import React, { StatelessComponent } from 'react';

import styles from './styles.css';

export interface SwitchProps {
  options: Array<[string]>;
  labels: Array<[string]>;
  name: string;
  isChecked?: boolean;
  disabled?: boolean;
  value: string;
  onChange: (value, event) => void;
}

const Switch: StatelessComponent<SwitchProps> = ({
  options,
  labels,
  onChange,
  name,
  isChecked,
  disabled,
  value,
}) => {
  const handleChange = (e: any): void => {
    const checked = e.target.checked;
    const checkedValue = !checked ? labels[0] : labels[1];
    if (onChange) {
      onChange({ value: checkedValue }, e);
    }
  };

  return (
    <div className="switch">
      <style jsx>{styles}</style>
      <label className="switch__wrapper">
        <input
          name={name}
          value={value}
          onChange={handleChange}
          className="switch__input"
          type="checkbox"
          checked={isChecked}
          disabled={disabled}
        />
        <span
          data-label-left={labels[0]}
          data-label-right={labels[1]}
          className="switch__label"
        >
          {options[0]}
        </span>
        <span
          data-label-left={labels[0]}
          data-label-right={labels[1]}
          className="switch__label"
        >
          {options[1]}
        </span>
        <div className="switch__icon" />
      </label>
    </div>
  );
};

export default Switch;
