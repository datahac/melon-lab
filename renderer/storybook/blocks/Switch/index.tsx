import React, { StatelessComponent } from 'react';

import styles from './styles.css';

export interface SwitchProps {
  disabled?: boolean;
  isChecked?: boolean;
  labels: string[];
  name: string;
  onChange: (e) => void;
  options: string[];
  value: string;
}

const Switch: StatelessComponent<SwitchProps> = ({
  disabled,
  isChecked,
  labels,
  name,
  onChange,
  options,
  value,
}) => {
  const handleChange = (e: any): void => {
    const checked = e.target.checked;
    const checkedValue = !checked ? labels[0] : labels[1];
    e.target.value = checkedValue;
    onChange(e);
  };

  return (
    <div className="switch">
      <style jsx={true}>{styles}</style>
      <label className="switch__wrapper">
        <input
          name={name}
          value={value}
          onChange={handleChange}
          className="switch__input"
          type="checkbox"
          defaultChecked={isChecked}
          disabled={disabled}
        />
        <span
          data-label-left={labels[0]}
          data-label-right={labels[1]}
          className="switch__label"
        >
          {options[0]}
        </span>
        <span
          data-label-left={labels[0]}
          data-label-right={labels[1]}
          className="switch__label"
        >
          {options[1]}
        </span>
        <div className="switch__icon" />
      </label>
    </div>
  );
};

export default Switch;
