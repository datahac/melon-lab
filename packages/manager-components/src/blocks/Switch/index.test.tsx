import React from 'react';
import Switch from './index';

const data = {
  options: ['ETH-T-M', 'MLN-T-M'],
  labels: ['Buy', 'Sell'],
  name: 'name',
  value: 'value',
  onChange: () => null,
};

describe('Switch', () => {
  const defaultElement = <Switch {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
