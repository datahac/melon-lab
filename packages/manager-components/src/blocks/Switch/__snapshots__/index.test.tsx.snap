// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render correctly 1`] = `
<div
  className="jsx-__scopedHash switch"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <label
    className="jsx-__scopedHash switch__wrapper"
  >
    <input
      className="jsx-__scopedHash switch__input"
      name="name"
      onChange={[Function]}
      type="checkbox"
      value="value"
    />
    <span
      className="jsx-__scopedHash switch__label"
      data-label-left="Buy"
      data-label-right="Sell"
    >
      ETH-T-M
    </span>
    <span
      className="jsx-__scopedHash switch__label"
      data-label-left="Buy"
      data-label-right="Sell"
    >
      MLN-T-M
    </span>
    <div
      className="jsx-__scopedHash switch__icon"
    />
  </label>
</div>
`;
